components/GroupProfilePanels/GroupSettingsPanel.js

```diff
@@ -0,0 +1,245 @@
+    Private Methods
+  \***************************************************************************/
+
+  _handleAddressChange (value) {
+    this._setChanges('address', value.formatted_address)
+  }
+
+  _handleChange ({ target }) {
+    const {
+      name,
+      validity,
+      value,
+    } = target
+
+    this._setChanges(name, value, validity.valid)
+  }
+
+  _handleSlugChange (event) {
+    this._setChanges('slug', convertStringToSlug(event.target.value))
+  }
+
+  async _handleSubmit (event) {
+    const {
+      group,
+      updateGroup,
+    } = this.props
+    const { changes } = this.state
+
+    event.preventDefault()
+
+    this.setState({ submitting: true })
+
+    await updateGroup(group.id, changes)
+
+    window.location.reload()
+  }
+
+  _isValid () {
+    const { validity } = this.state
+
+    return !Object.values(validity).includes(false)
+  }
+
+  _setChanges (key, value, isValid = true) {
+    const {
+      changes,
+      validity,
+    } = this.state
+
+    this.setState({
+      changes: {
+        ...changes,
+        [key]: value,
+      },
+      validity: {
+        ...validity,
+        [key]: isValid,
+      },
+    })
+  }
+
+
+
+
+
+  /***************************************************************************\
+    Public Methods
+  \***************************************************************************/
+
+  constructor (props) {
+    super(props)
+
+    this._bindMethods([
+      '_handleAddressChange',
+      '_handleChange',
+      '_handleSlugChange',
+      '_handleSubmit',
+    ])
+
+    this.state = {
+      changes: {},
+      submitting: false,
+      validity: {
+        address: true,
+        description: true,
+        discoverable: true,
+        name: true,
+        slug: true,
+      },
+    }
+  }
+
+  render () {
+    const { group } = this.props
+    const {
+      changes,
+      submitting,
+    } = this.state
+
+    const address = changes.address || group.attributes.address
+    const description = changes.description || group.attributes.description
+    // const discoverable = changes.discoverable || group.attributes.discoverable
+    const name = changes.name || group.attributes.name
+    // const slug = changes.slug || group.attributes.slug
+
+    return (
+      <section className="settings">
+        <form onSubmit={this._handleSubmit}>
+          <fieldset>
+            <label htmlFor="name">
+              Group name
+            </label>
+
+            <ValidatedInput
+              data-pattern-explainer="Please make sure to only use letters, numbers, hyphens (-), and underscores (_)"
+              disabled={submitting}
+              id="name"
+              name="name"
+              onChange={this._handleChange}
+              pattern="[\w\s_-]+"
+              placeholder="e.g. Quigley's Tavern"
+              required
+              type="text"
+              value={name} />
+          </fieldset>
+
+          {/* <fieldset>
```

Comment:
Was this intentionally left commented out?

Reply:
It was. The API doesn't support changing the permalink yet. When it does, we can just uncomment this component.